community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/steps/IndexSeekLeafPlanner.scala

```diff
@@ -19,29 +19,68 @@
  */
 package org.neo4j.cypher.internal.compiler.v3_1.planner.logical.steps
 
-import org.neo4j.cypher.internal.compiler.v3_1.commands.{SingleQueryExpression, QueryExpression}
+import org.neo4j.cypher.internal.compiler.v3_1.commands.{QueryExpression, SingleQueryExpression}
 import org.neo4j.cypher.internal.compiler.v3_1.planner.QueryGraph
+import org.neo4j.cypher.internal.compiler.v3_1.planner.logical.LeafPlansForVariable.maybeLeafPlans
 import org.neo4j.cypher.internal.compiler.v3_1.planner.logical._
 import org.neo4j.cypher.internal.compiler.v3_1.planner.logical.plans._
 import org.neo4j.cypher.internal.frontend.v3_1.SemanticTable
 import org.neo4j.cypher.internal.frontend.v3_1.ast._
 import org.neo4j.cypher.internal.frontend.v3_1.notification.IndexLookupUnfulfillableNotification
 import org.neo4j.kernel.api.index.IndexDescriptor
 
-abstract class AbstractIndexSeekLeafPlanner extends LeafPlanner {
+abstract class AbstractIndexSeekLeafPlanner extends LeafPlanner with LeafPlanFromExpression {
 
-  def apply(qg: QueryGraph)(implicit context: LogicalPlanningContext): Seq[LogicalPlan] = {
-    implicit val semanticTable = context.semanticTable
-    val predicates: Seq[Expression] = qg.selections.flatPredicates
+  override def producePlanFor(e: Expression, qg: QueryGraph)(implicit context: LogicalPlanningContext): Option[LeafPlansForVariable] = {
+    val arguments: Set[Variable] = qg.argumentIds.map(n => Variable(n.name)(null))
```

Comment:
I think we can use `InputPosition.NONE` here (although it seems to be `null` as well).